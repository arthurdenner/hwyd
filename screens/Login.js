/* @flow */
import React, { Component } from 'react';
import {
  StatusBar,
  Text,
  View,
} from 'react-native';
import styled from 'styled-components';
import firebase from 'firebase';

import {
  OLD_GERANIUM
} from '../utils/constants';

import {
  BackgroundGradient,
  ButtonText,
  Loader,
  Input
} from '../components';

type State = {
  loading: boolean,
  email: string,
  password: string,
  error: ?string,
};


const Container = styled(View) `
  align-items: center;
  flex: 1;
  justify-content: center;
  padding: 0 10%;
`;

const Error = styled(Text) `
  color: ${OLD_GERANIUM}
  margin: 20px;
`

export default class Login extends Component<*, State> {

  state = { email: '', password: '', error: '', loading: false };

  register(email: string = '', password: string = '') {
    firebase.auth().createUserWithEmailAndPassword(email, password)
      .then(() => { this.setState({ error: '', loading: false }); })
      .catch((error) => {
        this.setState({ error: error.message, loading: false });
      });
  }

  onLoginPress() {
    this.setState({ error: '', loading: true });

    const { email, password } = this.state;

    firebase.auth().signInWithEmailAndPassword(email, password)
      .then(() => this.setState({ loading: false }))
      .catch(() => this.register(email, password));
  }


  renderButtonOrSpinner() {
    if (this.state.loading) {
      return <Loader style={{ width: 50, height: 50 }} />
    }
    return (
      <ButtonText.Default onPress={this.onLoginPress.bind(this)}>
        <Text>Log in/Register</Text>
      </ButtonText.Default>
    )
  }

  render() {
    const { loading } = this.state
    return (
      <Container>
        <StatusBar barStyle="light-content" backgroundColor={OLD_GERANIUM} />
        <BackgroundGradient />
        <Input
          label='Email Address'
          type="email"
          placeholder='you@domain.com'
          value={this.state.email}
          onChangeText={email => this.setState({ email })}
        />
        <Input
          label='Password'
          type="password"
          autoCorrect={false}
          placeholder='*******'
          secureTextEntry
          value={this.state.password}
          onChangeText={password => this.setState({ password })}
        />
        <Error>{this.state.error}</Error>
        <ButtonText.Default type="submit" onPress={this.onLoginPress.bind(this)}>
          {loading ? (
            <Loader style={{ width: 50, height: 50 }} />
          ) : <Text>Log in/Register</Text>
          }
        </ButtonText.Default>
      </Container >
    );
  }
}
