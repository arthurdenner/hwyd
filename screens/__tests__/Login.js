import React from 'react';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import Login from '../Login';
import theme from '../../utils/theme';

describe('<Login />', () => {
  it('renders', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Login />
      </ThemeProvider>,
    );

    const root = tree.find(Login).shallow();
    expect(root).toMatchSnapshot();
  });
});
