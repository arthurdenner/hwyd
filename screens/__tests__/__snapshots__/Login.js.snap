// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders 1`] = `
<Styled(View)>
  <StatusBar
    animated={false}
    backgroundColor="#cf6a87"
    barStyle="light-content"
    showHideTransition="fade"
  />
  <BackgroundGradient />
  <Input
    label="Email Address"
    onChangeText={[Function]}
    placeholder="you@domain.com"
    type="email"
    value=""
  />
  <Input
    autoCorrect={false}
    label="Password"
    onChangeText={[Function]}
    placeholder="*******"
    secureTextEntry={true}
    type="password"
    value=""
  />
  <Styled(Text) />
  <Styled(Text)
    onPress={[Function]}
    type="submit"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Log in/Register
    </Text>
  </Styled(Text)>
</Styled(View)>
`;
